python/lsst/ts/standardscripts/auxtel/detector_characterization/get_std_flat_dataset.py

```diff
@@ -0,0 +1,279 @@
+                                           atspec=salobj.Remote(SALPY_ATSpectrograph,
+                                                                include=["changeFilter",
+                                                                         "changeDisperser",
+                                                                         "moveLinearStage"])))
+
+        # Had to increase this timeout to accommodate some issue with the camera.
+        self.cmd_timeout = 120.
+        self.read_out_time = 2.
+        self.n_dark = 10
+        self.t_dark = 400.
+        self.n_bias = 10
+        self.n_flat = 2
+        self.flat_base_exptime = 0.5
+        self.flat_dn_range = np.array([1., 2., 4., 8., 16., 32., 64., 128.])
+        self.filter = None
+        self.grating = None
+        self.linear_stage = None
+
+        # FIXME: Get this parameter from the camera configuration once late joiner is working
+        # on the open network.
+        self.maximum_exp_time = 401.  # Maximum exposure time in seconds.
+
+    async def configure(self,
+                        n_dark=None,
+                        t_dark=None,
+                        n_bias=None,
+                        n_flat=None,
+                        flat_base_exptime=None,
+                        flat_dn_range=None,
+                        filter_id=None,
+                        grating_id=None,
+                        linear_stage=None,
+                        read_out_time=2.):
+        """Configure script.
+
+        Parameters
+        ----------
+        n_dark : `int`
+            Number of dark images to be taken (default=10).
+        t_dark : `float`
+            Exposure time for the dark images in seconds (default=400s).
+        n_bias : `int`
+            Number of bias images to be taken (default=10).
+        n_flat : `int`
+            Number of flat field images to take for each exposure time.
+        flat_base_exptime : `float`
+            The exposure time for the first flat field image set (in seconds).
+        flat_dn_range : `list(float)`
+            List of values that multiply the `flat_base_exptime`. For instance, if you want to take
+            two sets of flat with exposure times `flat_base_exptime, 2*flat_base_exptime, 4*flat_dn_range`,
+            `flat_dn_range = [1., 2., 4.]`. Default is [1., 2., 4., 8., 16., 32., 64.].
+        filter_id : `int`
+            Id of the filter to use in the ATSpectrograph. If `None` (default) will ignore the setup step.
+        grating_id : `int`
+            Id of the grating to use in the ATSpectrograph. If `None` (default) will ignore the setup step.
+        linear_stage : `float`
+            Position of the linear stage in the ATSpectrograph. If `None` (default) will ignore the
+            setup step.
+        read_out_time :`float`
+            The read out time of the camera, in seconds. Used to estimate script duration.
```

Comment:
Since this is a characterization script, we may be in a situation where the readout time is changed to evaluate the impact on the CCD. So, I think it would be a good idea to leave it as an option to the user to set it up. For most cases, the user is probably ok with leaving it alone though.